libyul/backends/evm/ControlFlowGraphBuilder.cpp

```diff
@@ -85,6 +77,71 @@ std::unique_ptr<CFG> ControlFlowGraphBuilder::build(
 		cxx20::erase_if(node->entries, [&](CFG::BasicBlock* entry) -> bool {
 			return !reachabilityCheck.visited.count(entry);
 		});
+}
+// Sets the ``recursive`` member to ``true`` for all recursive function calls.
+void markRecursiveCalls(CFG& _cfg)
+{
+	map<CFG::BasicBlock*, vector<CFG::FunctionCall*>> callsPerBlock;
+	auto const& findCalls = [&](CFG::BasicBlock* _block)
+	{
+		if (auto* calls = util::valueOrNullptr(callsPerBlock, _block))
+			return *calls;
+		vector<CFG::FunctionCall*>& calls = callsPerBlock[_block];
+		util::BreadthFirstSearch<CFG::BasicBlock*>{{_block}}.run([&](CFG::BasicBlock* _block, auto _addChild) {
+			for (auto& operation: _block->operations)
+				if (auto* functionCall = get_if<CFG::FunctionCall>(&operation.operation))
+					calls.emplace_back(functionCall);
+			std::visit(util::GenericVisitor{
+				[&](CFG::BasicBlock::MainExit const&) {},
+				[&](CFG::BasicBlock::Jump const& _jump)
+				{
+					_addChild(_jump.target);
+				},
+				[&](CFG::BasicBlock::ConditionalJump const& _conditionalJump)
+				{
+					_addChild(_conditionalJump.zero);
+					_addChild(_conditionalJump.nonZero);
+				},
+				[&](CFG::BasicBlock::FunctionReturn const&) {},
+				[&](CFG::BasicBlock::Terminated const&)	{},
+			}, _block->exit);
+		});
+		return calls;
+	};
+	for (auto& functionInfo: _cfg.functionInfo | ranges::views::values)
+		for (CFG::FunctionCall* call: findCalls(functionInfo.entry))
+		{
+			util::BreadthFirstSearch<CFG::FunctionCall*> breadthFirstSearch{{call}};
+			breadthFirstSearch.run([&](CFG::FunctionCall* _call, auto _addChild) {
+				auto& calledFunctionInfo = _cfg.functionInfo.at(&_call->function.get());
+				if (&calledFunctionInfo == &functionInfo)
+				{
+					call->recursive = true;
+					breadthFirstSearch.abort();
+					return;
+				}
+				for (CFG::FunctionCall* recursiveCalls: findCalls(_cfg.functionInfo.at(&_call->function.get()).entry))
```

Comment:
These are not necessarily `recursive` I guess?

Reply:
Yeah, bad name indeed. More like ``nested``.